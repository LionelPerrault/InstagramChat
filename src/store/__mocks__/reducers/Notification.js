export default {
  unreadNotifications: 5,
  notifications: [
    {
      notify_id: 56,
      notify_by: 7,
      notify_by_username: 'ghalib',
      notify_time: '1480114098767',
      type: 'follow',
      user: 0,
      user_username: '',
      post_id: 0,
      group_id: 0,
      isFollowing: false,
    },
    {
      notify_id: 57,
      notify_by: 7,
      notify_by_username: 'ghalib',
      notify_time: '1480114098767',
      type: 'recommend',
      user: 8,
      user_username: 'coldplay',
      post_id: 0,
      group_id: 0,
      isFollowing: false,
    },
    {
      notify_id: 58,
      notify_by: 7,
      notify_by_username: 'ghalib',
      notify_time: '1480114098767',
      type: 'like',
      user: 0,
      user_username: '',
      post_id: 27,
      group_id: 0,
      isFollowing: false,
    },
    {
      notify_id: 59,
      notify_by: 7,
      notify_by_username: 'ghalib',
      notify_time: '1480114098767',
      type: 'invite',
      user: 0,
      user_username: '',
      post_id: 0,
      group_id: 11,
      isFollowing: false,
    },
  ],
}
