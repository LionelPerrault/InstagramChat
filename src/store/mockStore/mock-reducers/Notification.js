export default {
  unreadNotifications: 5
}
