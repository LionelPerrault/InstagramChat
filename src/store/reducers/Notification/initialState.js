export default {
  notifications: [],
  unreadNotifications: 0,
}
