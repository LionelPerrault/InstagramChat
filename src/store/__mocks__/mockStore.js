import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

// mock-reducers
import User from './reducers/User'
import Notification from './reducers/Notification'
import Group from './reducers/Group'
import Follow from './reducers/Follow'
import Explore from './reducers/Explore'
import Post from './reducers/Post'
import Hashtag from './reducers/Hashtag'
import Setting from './reducers/Setting'
import Message from './reducers/Message'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore,
  Post,
  Hashtag,
  Setting,
  Message,
}

const middlewares = [thunk]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
