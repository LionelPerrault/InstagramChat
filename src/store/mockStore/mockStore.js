import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import logger from 'redux-logger'
import promise from 'redux-promise-middleware'

// mock-reducers
import User from './mock-reducers/User'
import Notification from './mock-reducers/Notification'
import Group from './mock-reducers/Group'
import Follow from './mock-reducers/Follow'
import Explore from './mock-reducers/Explore'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore
}

const middlewares = [
  promise(), thunk, logger
]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
