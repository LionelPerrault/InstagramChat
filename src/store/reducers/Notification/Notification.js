/* eslint indent:0 */
/* eslint no-unreachable:0 */

import initialState from './initialState'

export default (state = initialState, action) => {
  let py = action.payload

  switch (action.type) {
    case 'GET_NOTIFICATIONS':
      return { ...state, notifications: py }
      break

    case 'CLEAR_NOTIFICATIONS':
      return { ...state, notifications: [] }
      break

    case 'GET_UNREAD_NOTIFICATIONS':
      return { ...state, unreadNotifications: py }
      break

    case 'READ_NOTIFICATIONS':
      return { ...state, unreadNotifications: 0 }
      break
  }

  return state
}
