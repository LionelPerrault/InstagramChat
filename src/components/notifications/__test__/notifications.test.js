import React from 'react'
import { Provider } from 'react-redux'
import mockStore from '../../../store/__mocks__/mockStore'
import Notifications, { PureNotifications } from '../notifications'
import { create } from 'react-test-renderer'
import Notification from '../../../store/__mocks__/reducers/Notification'
import { shallow } from 'enzyme'

describe('Notifications Component', () => {
  const comp = (
    <Provider store={mockStore}>
      <Notifications />
    </Provider>
  )

  it('should match snapshot', () => {
    const tree = create(comp).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should match snapshot when notifications.length == 0', () => {
    Notification.notifications = []
    const tree = create(comp).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should hide spinner when state.loading = false', () => {
    const wrapper = shallow(
      <PureNotifications
        dispatch={jest.fn()}
        notifications={Notification.notifications}
      />
    )
    wrapper.setState({ loading: false })
    expect(wrapper.find('IsLoading').prop('loading')).toBe(false)
  })
})
