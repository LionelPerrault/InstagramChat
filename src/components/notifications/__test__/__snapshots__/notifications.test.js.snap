// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications Component should match snapshot 1`] = `
<div>
  <div
    className="page_loading"
  >
    <div
      className="spinner"
    >
      <span />
      <span />
      <span />
    </div>
  </div>
  <div
    className="cLoading"
    style={
      Object {
        "animation": "bcCCNc 300ms ease 0s 1 normal none running",
        "backfaceVisibility": "visible",
      }
    }
  >
    <div
      className="notifications_div"
    >
      <div
        className="notifications_header"
      >
        <span
          className="noti_count"
        >
          4
           notifications
        </span>
        <div>
          <span
            className="clear_noti"
            data-tip="Clear notifications"
            onClick={[Function]}
          >
            <i
              className="material-icons"
            >
              clear_all
            </i>
          </span>
          <div
            className="__react_component_tooltip place-top type-dark "
            data-id="tooltip"
          >
            
          </div>
        </div>
      </div>
      <div
        className="noti follow_noti"
      >
        <img
          className="noti_avatar"
          src="/users/7/avatar.jpg"
        />
        <div
          className="noti_left"
        >
          <Link
            className="noti_bold noti_username"
            to="/profile/ghalib"
          >
            ghalib
          </Link>
          <span>
             started following you
          </span>
          <span
            className="noti_time"
          >
            a month ago
          </span>
        </div>
        <div
          className="noti_right follow_noti_right"
        >
          <a
            className="pri_btn follow"
            href="#"
            onClick={[Function]}
          >
            Follow
          </a>
        </div>
      </div>
      <div
        className="noti follow_noti"
      >
        <img
          className="noti_avatar"
          src="/users/7/avatar.jpg"
        />
        <div
          className="noti_left"
        >
          <Link
            className="noti_bold noti_username"
            to="/profile/ghalib"
          >
            ghalib
          </Link>
          <span>
             recommended coldplay to you
          </span>
          <span
            className="noti_time"
          >
            a month ago
          </span>
        </div>
        <div
          className="noti_right follow_noti_right"
        >
          <Link
            className="pri_btn"
            to="/profile/coldplay"
          >
            View coldplay
          </Link>
        </div>
      </div>
      <div
        className="noti follow_noti"
      >
        <img
          className="noti_avatar"
          src="/users/7/avatar.jpg"
        />
        <div
          className="noti_left"
        >
          <Link
            className="noti_bold noti_username"
            to="/profile/ghalib"
          >
            ghalib
          </Link>
          <span>
             liked your post
          </span>
          <span
            className="noti_time"
          >
            a month ago
          </span>
        </div>
        <div
          className="noti_right follow_noti_right"
        >
          <Link
            className="pri_btn"
            to="/post/27"
          >
            Open post
          </Link>
        </div>
      </div>
      <div
        className="noti follow_noti"
      >
        <img
          className="noti_avatar"
          src="/users/7/avatar.jpg"
        />
        <div
          className="noti_left"
        >
          <Link
            className="noti_bold noti_username"
            to="/profile/ghalib"
          >
            ghalib
          </Link>
          <span>
             invited to a group
          </span>
          <span
            className="noti_time"
          >
            a month ago
          </span>
        </div>
        <div
          className="noti_right follow_noti_right"
        >
          <Link
            className="pri_btn"
            to="/group/11"
          >
            View group
          </Link>
        </div>
      </div>
      <div
        className="page_end"
        onClick={[Function]}
      >
        <span>
          Looks like you've reached the end
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Notifications Component should match snapshot when notifications.length == 0 1`] = `
<div>
  <div
    className="page_loading"
  >
    <div
      className="spinner"
    >
      <span />
      <span />
      <span />
    </div>
  </div>
  <div
    className="cLoading"
    style={
      Object {
        "animation": "bcCCNc 300ms ease 0s 1 normal none running",
        "backfaceVisibility": "visible",
      }
    }
  >
    <div
      className="notifications_div"
    >
      <div
        className="notifications_header"
      >
        <span
          className="noti_count"
        >
          No
           notifications
        </span>
      </div>
      <div
        className="home_last_mssg"
        style={
          Object {
            "border": "",
          }
        }
      >
        <img
          src="/images/large.jpg"
        />
        <span
          className="nothingMssg"
        >
          You have no notifications!!
        </span>
        <span
          className="secondMssg"
        >
          
        </span>
      </div>
    </div>
  </div>
</div>
`;
